crates/wasi-common/src/sys/windows/hostcalls_impl/fs.rs

```diff
@@ -428,43 +427,46 @@ pub(crate) fn path_rename(resolved_old: PathGet, resolved_new: PathGet) -> Resul
 
     // TODO handle symlinks
 
-    fs::rename(&old_path, &new_path).or_else(|e| match e.raw_os_error() {
-        Some(e) => {
-            use winx::winerror::WinError;
-
-            log::debug!("path_rename at rename error code={:?}", e);
-            match WinError::from_u32(e as u32) {
-                WinError::ERROR_ACCESS_DENIED => {
-                    // So most likely dealing with new_path == dir.
-                    // Eliminate case old_path == file first.
-                    if old_path.is_file() {
-                        Err(Error::EISDIR)
-                    } else {
-                        // Ok, let's try removing an empty dir at new_path if it exists
-                        // and is a nonempty dir.
-                        fs::remove_dir(&new_path)
-                            .and_then(|()| fs::rename(old_path, new_path))
-                            .map_err(Into::into)
+    if let Err(err) = fs::rename(&old_path, &new_path) {
```

Comment:
FWIW this'd probably be a good place to do something like:

```rust
let err = match fs::rename(...) {
    Ok(()) => return Ok(()),
    Err(e) => e,
};
// de-indented error-handling code

Reply:
Or even better yet:

```rust
fn extract_raw_os_error(e: io::Error) -> Result<i32> {
    // match returning EIO for "other errors"
}

let err = match fs::rename(...) {
    Ok(()) => return Ok(()),
    Err(e) => extract_raw_os_error(e)?,
};
// de-indented error-handling code

Reply:
Nice, I like it! I'll try and enforce this principle whenever I write similar code in the future. Also, while I won't do this in this PR, when I'm less creative I might scan `wasi-common` and related, and rewrite any suitable occurrence using this principle. :+1: